docs/api/TestUtils.md

```diff
@@ -0,0 +1,194 @@
+
+When `executeAction` is called, it will push an object to the `executeActionCalls` array. Each object contains an `action` and `payload` key.
+
+`getStore` calls will be proxied to a dispatcher instance, which you can register stores to via `MockActionContext.registerStore(MockStore)`.
+
+### Usage
+
+Here is an example mocha test that display using each of `ActionContext` methods being tested:
+
+```js
+var MockActionContext = require('./utils').createMockActionContext();
+
+// Real store, overridden with MockStore in test
+var createStore = require('fluxible/addons').createStore;
+var FooStore = createStore({
+    storeName: 'FooStore'
+});
+
+// Actions being tested
+var myAction = function (actionContext, payload, done) {
+    var foo = actionContext.getStore(FooStore).getFoo() + payload;
+    actionContext.dispatch('FOO', foo);
+    actionContext.executeAction(otherAction, foo, done);
+};
+
+var otherAction = function (actionContext, payload, done) {
+    done();
+};
+
+// Register the mock FooStore
+MockActionContext.registerStore(createStore({
+    storeName: 'FooStore', // Matches FooStore.storeName
+    handlers: {
+        FOO: 'handleFoo'
+    },
+    initialize: function () {
+        this.foo = 'foo';
+    },
+    handleFoo: function (payload) {
+        this.foo = payload
+    },
+    getFoo: function () {
+        return this.foo;
+    }
+}));
+
+
+// Tests
+describe('myAction', function () {
+    var assert = require('assert');
+    var actionContext;
+
+    beforeEach(function () {
+        actionContext = new MockActionContext();
+    });
+
+    it('should dispatch foo', function (done) {
+        myAction(actionContext, 'bar', function () {
+            assert.equal(1, actionContext.dispatchCalls.length);
+            assert.equal('FOO', actionContext.dispatchCalls[0].name);
+            assert.equal('foobar', actionContext.dispatchCalls[0].payload);
+            assert.equal(1, actionContext.executeActionCalls.length);
+            assert.equal(otherAction, actionContext.executeActionCalls[0].action);
+            assert.equal('foobar', actionContext.executeActionCalls[0].payload);
+            done();
+        });
+    });
+});
+```
+
+## MockComponentContext
+
+Useful for testing your components, you can pass an instance of `MockComponentContext` to your component to record the methods that the component calls on the context.
+
+When `executeAction` is called, it will push an object to the `executeActionCalls` array. Each object contains an `action` and `payload` key.
+
+`getStore` calls will be proxied to a dispatcher instance, which you can register stores to via `MockActionContext.registerStore(MockStore)`.
+
+### Usage
+
+Here is an example component test that uses `React.TestUtils` to render the component into `jsdom` to test the store integration.
+
+```js
+var MockComponentContext = require('./utils').createMockComponentContext();
```

Comment:
Should we update this to use `require('fluxible/utils').createMockActionContext();` instead of the relative path?